eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -19,10 +39,56 @@ steps:
         # Install gcc 9
         brew install gcc@9
         gcc --version
+    displayName: MacOS - Set tools versions
+    condition: >-
+      and(
+        succeeded(),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
+        contains(variables['OSVmImage'], 'macOS')
+      )
+
+  # Clear the vcpkg folder so the cache can successfully deploy to that location
+  - bash: |
+      id
+      sudo rm -rf $(VCPKG_INSTALLATION_ROOT)
+      sudo mkdir -p $(VCPKG_INSTALLATION_ROOT)
+      sudo chown `id --name -u`.`id --name -g` $(VCPKG_INSTALLATION_ROOT)
+    displayName: Linux - Clear vcpkg folder
+    condition: >-
+      and(
+        succeeded(),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
+        contains(variables['OSVmImage'], 'ubuntu')
+      )
 
+  # Attempt to restore vcpkg from the cache
+  - task: Cache@2
+    inputs:
+      key: >-
+        $(Agent.JobName)
+        | "${{ parameters.VcpkgVersion }}"
+        | $(Agent.Os)
+        | $(${{ parameters.DependenciesVariableName }})
+      path: $(VCPKG_INSTALLATION_ROOT)
+      cacheHitVar: VcpkgRestoredFromCache
+    displayName: Vcpkg Cache
+    condition: >-
+      and(
+        succeeded(),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
+        not(eq(variables['Skip.VcpkgCache'], 'true'))
+      )
+
+  # Install vcpkg on MacOS
+  - task: Bash@3
+    inputs:
+      targetType: inline
+      workingDirectory: $(Agent.TempDirectory)
+      script: |
         git clone https://github.com/Microsoft/vcpkg.git
```

Comment:
Now that we are always installing if not found in the cache is there still a need for a forked set of vcpkg bootstrap commands?